src/rp/index.ts

```diff
@@ -206,6 +219,51 @@ export class RelyingPartySolution {
     );
   }
 
+  /**
+   * Select a public key from those provided by the RP.
+   * Keys with algorithm "RSA-OAEP-256" or "RSA-OAEP" are expected, the firsts to be preferred.
+   *
+   * @param entity The RP entity configuration
+   * @returns A suitable public key with its compatible encryption algorithm
+   * @throws {NoSuitableKeysFoundInEntityConfiguration} If entity do not contain any public key suitable for encrypting
+   */
+  private choosePublicKeyToEncrypt(
+    entity: RpEntityConfiguration
+  ):
+    | { jwk: JWK & { alg: "RSA-OAEP-256" }; enc: "A256CBC-HS512" }
+    | { jwk: JWK & { alg: "RSA-OAEP" }; enc: "A128CBC-HS256" } {
+    const shuffle = () => (Math.random() > 0.5 ? 1 : -1);
+
+    // Look for keys using "RSA-OAEP-256", and pick a random one
+    const [usingRsa256] = entity.payload.jwks.keys
+      .filter(
+        <T>(k: T & { alg?: string }): k is T & { alg: "RSA-OAEP-256" } =>
+          typeof k.alg === "string" && k.alg === "RSA-OAEP-256"
+      )
+      .sort(shuffle);
+
+    if (usingRsa256) {
+      return { jwk: usingRsa256, enc: "A256CBC-HS512" };
```

Comment:
since encryption has nothing to do with the key what do you think about moving this logic outside?

```suggestion
      return usingRsa256;
```

and use a function like:
```js
getEncryptionAlgByJwk(jwk: JWK): "A256CBC-HS512" | "A128CBC-HS256"
```

Reply:
Fair enough, I will.